### DISCUSSION TITLE
Publishing from Codemagic to Firebase

### DISCUSSION THREAD

**Discussion by leadmonkeyboy** [codemagic-ci-cd/codemagic-docs#2817] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Run the Android Release Workflow configured in the codemagic.yaml file.

### Expected results

The AAB will be published to Email, Google Play and Firebase

### Actual results

Firebase publishing fails with the error: Error: failed to determine AAB info. HTTP Error: 403, The caller does not have permission.

I have tried the exact command locally to publish to firebase, firebase appdistribution:distribute /Users/builder/clone/build/app/outputs/bundle/prodRelease/app-prod-release.aab --app 1:346326001467:android:4b10eedd5f1e962ceab91e --groups android-testers and it works as expected. I'm sure this is an issue with the credentials, but I went through the integration steps more than once and can't figure out the issue. And advice would be appreciated. Thanks!

### Build id (optional)

672d27b18b75ffa2566cde12

**Comment by leadmonkeyboy** [comment]
Of course, right after I publish this I realize that the issue was related to how the service accounts were configured.

**Comment by EmanuelDavid** [comment]
I have the exact problem, what was wrong with your service account? roles or what?

**Comment by leadmonkeyboy** [comment]
Sorry, I don't remember the exact details, but yes, in my case it was a permission issue with the service account.

**Reply by EmanuelDavid** [reply]
k
